include/deal.II/base/memory_consumption.h

```diff
@@ -316,6 +341,25 @@ namespace MemoryConsumption
 
 
 
+  template <typename T, std::size_t N>
+  std::size_t memory_consumption (const std_cxx11::array<T,N> &v)
+  {
+    // shortcut for types that do not allocate memory themselves
+    if (std_cxx11::is_fundamental<T>::value || std_cxx11::is_pointer<T>::value)
+      {
+        return sizeof(v);
+      }
+    else
+      {
+        std::size_t mem = sizeof(std_cxx11::array<T,N>);
+        for (std::size_t i=0; i!=N; ++i)
```

Comment:
No. I don't know. I can change it if you want me to. (I think I was thinking that `std::size_t` is unsigned, in which case you need to write `!=` of `N` could be zero.